addons/google_calendar/utils/google_calendar.py

```diff
@@ -70,6 +70,29 @@ def get_events(self, sync_token=None, token=None, event_id=None, timeout=TIMEOUT
 
         return GoogleEvent(events), next_sync_token, default_reminders
 
+    @requires_auth_token
+    def fetch_working_location_events(self, token=None, start=None, end=None, timeout=3):
+        """ Get working location events from Google Calendar. """
+        url = "/calendar/v3/calendars/primary/events"
+        headers = {'Content-type': 'application/json'}
+        params = {
+            'access_token': token,
+            'timeMin': start.isoformat() + 'Z',
+            'timeMax': end.isoformat() + 'Z',
+            'eventTypes': ['workingLocation'],
+        }
```

Comment:
it will crash if it called without start or end value provided.
I would default to today in that case.